backend/catalogue/management/commands/generate_layer_previews.py

```diff
@@ -78,42 +92,58 @@ def retrieve_image(layer):
 class Command(BaseCommand):
     def add_arguments(self, parser):
         parser.add_argument(
-            '--layer_id'
+            '--layer_id',
+            help='Specify a layer ID for the layer preview image you want to generate'
+        )
+        parser.add_argument(
+            '--only_generate_missing',
+            help='If true, skips generating images for layers where images already exist'
         )
 
     def handle(self, *args, **options):
-        layer_id = int(options['layer_id']
-                       ) if options['layer_id'] != None else None
+        layer_id = int(options['layer_id']) if options['layer_id'] != None else None
+        only_generate_missing = options['only_generate_missing'].lower() in ['t', 'true'] if options['only_generate_missing'] != None else False
+
+
 
         if layer_id is not None:
             layer = Layer.objects.get(id=layer_id)
 
-            bytes_io = BytesIO()
+            filepath = f'layer_previews/{layer.id}.png'
+            if not only_generate_missing or not default_storage.exists(filepath):
+                bytes_io = BytesIO()
```

Comment:
I'd recommend using a with-block for bytesio (just avoids traps with cleanup, etc).  Bit of a dry reference, but these: https://docs.python.org/3/reference/compound_stmts.html#with

Reply:
Ah, very useful to know! I guess I should have been more curious about that syntax when it was being used for the database connection. I've used it as suggested.